packages/@react-aria/overlays/src/useModal.tsx

```diff
@@ -37,7 +37,7 @@ const Context = React.createContext<ModalContext | null>(null);
 export function ModalProvider(props: ModalProviderProps) {
   let {children} = props;
   let parent = useContext(Context);
-  let [modalCount, setModalCount] = useState(parent ? parent.modalCount : 0);
+  let [modalCount, setModalCount] = useState(0);
```

Comment:
This fixes the issue with `aria-hidden` on an open dialog that came up before. I believe this logic was simply incorrect: each ModalProvider keeps track of the number of modals open in its subtree, but this seems to store a count of *all* open modals. 🤷